pkg/GAPJulia/JuliaInterface/gap/convert.gd

```diff
@@ -4,15 +4,503 @@
+#! gap> JuliaToGAP( IsList, l );
+#! [ 1, 2, 4 ]
+#! </Example>
+#!  The following values for <A>filt</A> are supported.
+#!  <Ref Filt="IsInt" BookName="ref"/>,
+#!  <Ref Filt="IsRat" BookName="ref"/>,
+#!  <Ref Filt="IsFFE" BookName="ref"/>,
+#!  <C>IsFloat</C> (see <Ref Chap="Floats" BookName="ref"/>),
+#!  <Ref Filt="IsBool" BookName="ref"/>,
+#!  <Ref Filt="IsChar" BookName="ref"/>,
+#!  <Ref Filt="IsRecord" BookName="ref"/>,
+#!  <Ref Filt="IsString" BookName="ref"/>,
+#!  <Ref Filt="IsRange" BookName="ref"/>,
+#!  <Ref Filt="IsBlist" BookName="ref"/>,
+#!  <Ref Filt="IsList" BookName="ref"/>.
+#!  See Section <Ref Sect="Section_Conversion_rules"/> for the admissible types
+#!  of <A>juliaobj</A> in these cases.
 DeclareConstructor("JuliaToGAP", [IsObject, IsObject]);
 
-#! TODO: write proper documentation comment
 DeclareConstructor("JuliaToGAP", [IsObject, IsObject, IsBool]);
 
-#! TODO: write proper documentation comment, for one and two argument variants
+#! @Arguments [juliatype, ]gapobj
+#! @Returns a &Julia; object
+#! @Description
+#!  Let <A>gapobj</A> be an object
+#!  for which a conversion to &Julia; is provided,
+#!  in the sense of Section <Ref Sect="Section_Conversion_rules"/>,
+#!  such that a corresponding &Julia; object with type <A>juliatype</A>
+#!  can be constructed.
+#!  Then <Ref Func="GAPToJulia"/> returns this &Julia; object.
+#!  <P/>
+#!  If <A>juliatype</A> is not given then a default type is chosen.
+#!  The function is implemented via the &Julia; function
+#!  <C>GAP.gap_to_julia</C>.
+#!  <P/>
+#! <!-- Note that the Julia output contains the "forbidden" sequence "]]>",
+#!      thus the CDATA syntax cannot be used. -->
+#! <Example>
+#! gap> GAPToJulia( 1 );
+#! 1
+#! gap> GAPToJulia( JuliaEvalString( "Rational{Int64}" ), 1 );
+#! &lt;Julia: 1//1>
+#! gap> l:= [ 1, 3, 4 ];;
+#! gap> GAPToJulia( l );
+#! &lt;Julia: Any[1, 3, 4]>
+#! gap> GAPToJulia( JuliaEvalString( "Array{Int,1}" ), l );
+#! &lt;Julia: [1, 3, 4]>
+#! gap> m:= [ [ 1, 2 ], [ 3, 4 ] ];;
+#! gap> GAPToJulia( m );
+#! &lt;Julia: Any[Any[1, 2], Any[3, 4]]>
+#! gap> GAPToJulia( JuliaEvalString( "Array{Int,2}" ), m );
+#! &lt;Julia: [1 2; 3 4]>
+#! gap> r:= rec( a:= 1, b:= [ 1, 2, 3 ] );;
+#! gap> GAPToJulia( r );
+#! &lt;Julia: Dict{Symbol,Any}(:a => 1,:b => Any[1, 2, 3])>
+#! </Example>
 DeclareGlobalFunction("GAPToJulia");
+
+#! @Section Open items
+#! <List>
+#! <Item>
+#!   On the &Julia; side,
+#!   an alternative to <C>gap_to_julia</C> would be a bunch of
+#!   constructor methods, e.g., <C>BigInt(x::MPtr)</C> or <C>big(x::MPtr)</C>
+#!   to convert a &GAP; object to a &Julia; <C>BigInt</C>, if possible.
+#!   Explain why we did not do this instead of <C>gap_to_julia</C>;
+#!   think about the possibility of still prodiving these,
+#!   implemented by calling <C>gap_to_julia</C> or vice versa.
+#!   (Pro constructors: <C>BigInt(x)</C> is shorter than
+#!   <C>gap_to_julia(BigInt, x)</C>.
+#!   Con constructors: symmetry perhaps?)
```

Comment:
Also: not (yet?) clear how to deal with recursive conversions that preserve identity. I.e. converting the GAP list  `[ ~ ]` which contains itself, or something like `[ "x", ~[1] ]` which contains the same (identical) string twice. Of course for many applications this is irrelevant, but maybe not for some others.

That said, I absolutely think we should offer these constructors, but retain `gap_to_julia` (at least for now, as people are using it, and it has the above mentioned recursive feature)